core/distributed/preconditioner/schwarz.cpp

```diff
@@ -77,10 +91,46 @@ void Schwarz<ValueType, LocalIndexType, GlobalIndexType>::apply_dense_impl(
     const VectorType* dense_b, VectorType* dense_x) const
 {
     using Vector = matrix::Dense<ValueType>;
+    using dist_vec = experimental::distributed::Vector<ValueType>;
     auto exec = this->get_executor();
-    if (this->local_solver_ != nullptr) {
+
+    // Two-level
+    if (this->coarse_solver_ != nullptr && this->coarse_level_ != nullptr) {
         this->local_solver_->apply(gko::detail::get_local(dense_b),
                                    gko::detail::get_local(dense_x));
```

Comment:
I think you can move the local_solver part to the beginning without another condition.
```
if (local_solver_ != nullptr) {...}
if (this->coarse_solver_ ...) {...}
```

Reply:
I prefer to keep them in separate code paths for better code readability.